src/components/views/EditUserPreferencePage.js

```diff
@@ -1,13 +1,209 @@
+import {api, handleError} from "../../helpers/api";
+import 'styles/views/LoginRegistration.scss'
+import Slider from "@mui/material/Slider";
+import FormLabel from '@mui/material/FormLabel';
+import FormControl from '@mui/material/FormControl';
+import FormGroup from '@mui/material/FormGroup';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import FormHelperText from '@mui/material/FormHelperText';
+import Checkbox from '@mui/material/Checkbox';
+
 
 /*
 Edit User Preferences Page
  */
+const EditUserPreferencePage = () =>{
+    const id = localStorage.getItem("id")
+
+    const history = useHistory();
+
+    const [agePreference, setAgePreference] = useState([1,100])
+    const [distance, setDistance] = useState([0,10000])
+    const [genderPreference, setGenderPreference] = React.useState({
+        MALE: true,
+        FEMALE: false,
+        OTHER: false,
+    });
+    const { MALE, FEMALE, OTHER } = genderPreference;
+    const noGenderSelectedError = [MALE, FEMALE, OTHER].filter((v) => v).length === 0;
+
+
+    //Gets all the preferences of the user
+    useEffect(() => {
+        async function getPreferences(){
+            try{
+                const response = await api.get('/users/'+id)
+                setGenderPreference({
+                    MALE: response.data.genderPreferences.includes("MALE"),
+                    FEMALE: response.data.genderPreferences.includes("FEMALE"),
+                    OTHER: response.data.genderPreferences.includes("OTHER"),
+                })
+                setAgePreference([response.data.minAge, response.data.maxAge])
+                //TODO: Add distance preferences for the user when GeoLocation API gets installed
+            } catch (error) {
+                console.error("Details:", error);
+                alert("Invalid Input:\n " + handleError(error));
+            }
+        }
+        getPreferences()
+    }, []);
+
+
+    //Method to change the age for ageRangeslider
+    const handleChangeAge = (event, newValue, activeThumb) => {
+        if (!Array.isArray(newValue)) {
+            return;
+        }
+
+        if (activeThumb === 0) {
+            setAgePreference([Math.min(newValue[0], agePreference[1]), agePreference[1]]);
+            console.log(agePreference)
+        } else {
+            setAgePreference([agePreference[0], Math.max(newValue[1], agePreference[0])]);
+            console.log(agePreference)
+        }
+    };
+
+    //Method to changes distances for distance range slider
+    const handleChangeDistance = (event, newValue, activeThumb) => {
+        if (!Array.isArray(newValue)) {
+            return;
+        }
+
+        if (activeThumb === 0) {
+            setDistance([Math.min(newValue[0], distance[1]), distance[1]]);
+            console.log(distance)
+        } else {
+            setDistance([distance[0], Math.max(newValue[1], distance[0])]);
+            console.log(distance)
+        }
+    };
```

Comment:
I think it makes sense if you implement this for the geolocation. Is there a way to make the slider not linear? because the smallest unit I can select now is 22 km. Also it would make more sense to remove the left knob, as it would be akward to want to match with people that are more than x km away from me. [for future]

Reply:
Yes, on the two knobs i am unsure, if we should move one. I will also find out if there is an option to scale it logarithmically, otherwise i would do the distance just by selecting a range with a radiobutton with predefined options or maybe user input. But i think should be doable with the rangeslider